compat/src/render.js

```diff
@@ -210,6 +210,12 @@ options.vnode = vnode => {
 			if (props.className != null) normalizedProps.class = props.className;
 			Object.defineProperty(normalizedProps, 'className', classNameDescriptor);
 		}
+	} else if (type.defaultProps != null) {
```

Comment:
I'm wondering if we should guard against type being undefined here? (`createElement(X.MissingProp)`)